src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs

```diff
@@ -160,7 +160,6 @@ private bool CanHandleLocally(KernelCommand command)
         switch (command)
         {
             case AnonymousKernelCommand:
-                return base.HandleAsync(command, context);
```

Comment:
this could lead to a deadlock if the anonymous command targets the proxy. For sure this will throw exception as it will try to forward the command on the sender and will fail at serialization. This looks like a test gap to pinpoint

Reply:
Yes this PR is fixing one specific case where we were serializing AnonymousKernelCommand (inside SubmissionParser.cs). But the larger test gap has not been addressed yet - I have included this in #2884 and plan to address that in a follow up PR.